### PR TITLE
UGENE-7247. Application unexpectedly closed after saving Workflow properties

### PR DESCRIPTION
**Pull request by evelinabis**
https://ugene.dev/tracker/browse/UGENE-7247

### CODE REVIEW

File: src/corelibs/U2Designer/src/wizard/WizardController.cpp [hunk 1]
```diff
@@ -66,11 +66,11 @@
 
 namespace U2 {
 
-WizardController::WizardController(Schema *s, const Wizard *w)
-    : QObject(), schema(s), wizard(w), runAfterApply(false) {
+WizardController::WizardController(std::shared_ptr<Schema> s, const Wizard *w)
```
Comment (yalgaer) [inline src/corelibs/U2Designer/src/wizard/WizardController.cpp]: This code does not compile
I would suggest:
1) Use QT variants of the pointers. They have no cross platform/compiler issues.
2) When use std always use std prefix (see move())
3) For any non-primitive type method parameters (even shared structures) use (const Type&).

**Review by yalgaer — CHANGES_REQUESTED** [review]

File: src/corelibs/U2Designer/src/wizard/WizardController.cpp [hunk 2]
```diff
@@ -66,11 +66,11 @@
...
+    : QObject(), schema(move(s)), wizard(w), runAfterApply(false) {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Designer/src/wizard/WizardController.cpp]: This thing looks a bit strange (I mean, `std::move`) - a bit over-designed or like you don't get the idea of the move semantic. 

It's difficult to explain why it's not really correct, I've tied to text a good comment, but it was very large and complicated in result. The main idea is: do not use `std::move` in this context, just replace your argument `std::shared_ptr<Schema> s` with `const std::shared_ptr<Schema>& s` (and it's better to use `QSharedPointer`)

File: src/corelibs/U2Lang/src/support/WorkflowIOTasks.cpp [hunk 3]
```diff
@@ -38,9 +38,9 @@ using namespace Workflow;
 /************************************
  * LoadWorkflowTask
  ************************************/
-LoadWorkflowTask::LoadWorkflowTask(Schema *s, Workflow::Metadata *m, const QString &u)
+LoadWorkflowTask::LoadWorkflowTask(std::shared_ptr<Schema> s, Workflow::Metadata *m, const QString &u)
     : Task(tr("Loading workflow"), TaskFlag_None),
-      url(u), schema(s), meta(m) {
+      url(u), schema(move(s)), meta(m) {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Lang/src/support/WorkflowIOTasks.cpp]: The same

File: src/corelibs/U2Lang/src/tasks/SchemaEstimationTask.cpp [hunk 4]
```diff
@@ -40,8 +40,8 @@ EstimationResult::EstimationResult() {
     cpuCount = -1;
 }
 
-SchemaEstimationTask::SchemaEstimationTask(const Schema *_schema, const Metadata *_meta)
-    : Task(tr("Workflow estimation task"), TaskFlag_None), schema(_schema), meta(_meta) {
+SchemaEstimationTask::SchemaEstimationTask(std::shared_ptr<const Schema> _schema, const Metadata *_meta)
+    : Task(tr("Workflow estimation task"), TaskFlag_None), schema(move(_schema)), meta(_meta) {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Lang/src/tasks/SchemaEstimationTask.cpp]: The same

File: src/corelibs/U2Lang/src/tasks/SchemaEstimationTask.cpp [hunk 5]
```diff
@@ -75,8 +75,8 @@ EstimationResult SchemaEstimationTask::result() const {
 /************************************************************************/
 /* Utils */
 /************************************************************************/
-ExtimationsUtilsClass::ExtimationsUtilsClass(QScriptEngine &_engine, const Schema *_schema)
-    : engine(_engine), schema(_schema) {
+ExtimationsUtilsClass::ExtimationsUtilsClass(QScriptEngine &_engine, std::shared_ptr<const Schema> _schema)
+    : engine(_engine), schema(move(_schema)) {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Lang/src/tasks/SchemaEstimationTask.cpp]: The same

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
Use Qt smart pointers and get rid of the move semantic features - it this context it's excessively

**Review by yalgaer — APPROVED** [review]

**Review by DmitriiSukhomlinov — APPROVED** [review]

**Pull request merged**